pkg/util/node/node.go

```diff
@@ -71,31 +71,26 @@ func GetPreferredNodeAddress(node *v1.Node, preferredAddressTypes []v1.NodeAddre
 // node, and the first IP is used as the `.status.HostIP` for all pods on the node.
 func GetNodeHostIPs(node *v1.Node) ([]net.IP, error) {
 	// Re-sort the addresses with InternalIPs first and then ExternalIPs
-	allIPs := make([]net.IP, 0, len(node.Status.Addresses))
+	var internalIPs, externalIPs []net.IP
```

Comment:
The old code pre-allocates `allIPs` to be the correct length, while your new code does not pre-allocate any of the three arrays, likely resulting in multiple reallocations.

Why exactly do you think the rewrite is an improvement?

Reply:
The main optimization is to reduce one iteration of `node.Status.Addresses`

Reply:
Thank you for your question, I have given a new optimization solution to the problem you mentioned, and tested it locally, please give me an `ok-to-test`, and request a code review. @danwinship

Reply:
reducing one iteration is not much of an optimization O(n-1) == O(n) 
and the code is much more complex IMHO

Reply:
> reducing one iteration is not much of an optimization O(n-1) == O(n) and the code is much more complex IMHO

Thanks for your question, I tried to further optimize the performance of the code to reduce space consumption and improve readability, the extra space consumption no longer increases with the length of `node.Status.Addresses`, but is fixed at 2

The focus here is on finding two different formats of ip, with priority given to `InternalIP`, so the search for `ExternalIP` only needs to find two different formats of ip

Request your code review. @aojea